app/src/main/java/com/framgia/englishconversation/widget/ZoomableImageView.java

```diff
@@ -0,0 +1,213 @@
+package com.framgia.englishconversation.widget;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Matrix;
+import android.graphics.PointF;
+import android.support.v7.widget.AppCompatImageView;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.ScaleGestureDetector;
+import android.view.View;
+
+public class ZoomableImageView extends AppCompatImageView {
+    Matrix matrix = new Matrix();
+
+    static final int NONE = 0;
+    static final int DRAG = 1;
+    static final int ZOOM = 2;
+    int mode = NONE;
+
+    PointF last = new PointF();
+    PointF start = new PointF();
+    float minScale = 1f;
+    float maxScale = 4f;
+    float[] m;
+
+    float redundantXSpace, redundantYSpace;
+    float width, height;
+    float saveScale = 1f;
+    float right, bottom, origWidth, origHeight, bmWidth, bmHeight;
+
+    ScaleGestureDetector scaleDetector;
+    Context context;
+
+
+
+    public ZoomableImageView(Context context, AttributeSet attr) {
+        super(context, attr);
+        super.setClickable(false);
+        this.context = context;
+        scaleDetector = new ScaleGestureDetector(context, new ScaleListener());
+        matrix.setTranslate(1f, 1f);
+        m = new float[9];
+        setImageMatrix(matrix);
+        setScaleType(ScaleType.MATRIX);
+
+        setOnTouchListener(new OnTouchListener() {
+
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
```

Comment:
WARNING: [ClickableViewAccessibility] `com/framgia/englishconversation/widget/ZoomableImageView$1#onTouch` should call `View#performClick` when a click is detected
 If a `View` that overrides `onTouchEvent` or uses an `OnTouchListener` does not also implement `performClick` and call it when clicks are detected, the `View` may not handle accessibility actions properly. Logic handling the click actions should ideally be placed in `View#performClick` as some accessibility services invoke `performClick` when a click action should occur.